ethcore/light/src/transaction_queue.rs

```diff
@@ -358,11 +364,26 @@ impl TransactionQueue {
 		self.listeners.push(f);
 	}
 
+	/// Add a transaction queue listener.
+	pub fn tx_statuses_receiver(&mut self) -> mpsc::UnboundedReceiver<Arc<Vec<(H256, TxStatus)>>> {
+		let (sender, receiver) = mpsc::unbounded();
+		self.tx_statuses_listeners.push(sender);
+		receiver
+	}
+
 	/// Notifies all listeners about new pending transaction.
-	fn notify(&self, hashes: &[H256]) {
+	fn notify(&mut self, hashes: &[H256], status: TxStatus) {
 		for listener in &self.listeners {
 			listener(hashes)
 		}
+
+		let to_send: Arc<Vec<(H256, TxStatus)>> = Arc::new(
```

Comment:
there's no need for `Arc` here.

`Vec<T>` implements `Clone` if `T: Clone`.

`T` here is a tuple, `(H256, TxStatus)` which also implements clone because `H256` and `TxStatus` implements `Clone`


Nice work overall

Reply:
@seunlanlege Then here, too, do not need Arc?
https://github.com/paritytech/parity-ethereum/pull/10558/files/286ba53d13330a39a3fc66b2053e2ab9d5eb058c#diff-f34b6331b91a5d4893e05107b8ee0622R124
https://github.com/paritytech/parity-ethereum/pull/10558/files/286ba53d13330a39a3fc66b2053e2ab9d5eb058c#diff-f34b6331b91a5d4893e05107b8ee0622R140

Reply:
We do need `Arc`. There is a difference between cloning an `Arc` reference and the entire vector that might be huge.

Reply:
@tomusdrw yeah you're right. Didn't think of that.